flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/data/RowDataProjection.java

```diff
@@ -134,11 +176,26 @@ private static RowData.FieldGetter createFieldGetter(RowType rowType,
 
   public RowData wrap(RowData row) {
     this.rowData = row;
+    this.valueCache.clear();
     return this;
   }
 
   private Object getValue(int pos) {
-    return getters[pos].getFieldOrNull(rowData);
+    return valueCache.computeIfAbsent(pos, key -> {
+      Object value = null;
+      if (projectedFields != null) {
+        int[] projectedField = projectedFields[key];
+        for (int index : projectedField) {
```

Comment:
Drill down.

Reply:
Is this correct ?  we are replacing the previous value with the next pos value...